modules/service/src/main/scala/lucuma/odb/logic/Generator.scala

```diff
@@ -475,17 +517,7 @@ object Generator {
           def sequenceDigest(s: Stream[Pure, Atom[D]]): Either[OdbError, SequenceDigest] =
             s.fold(SequenceDigest.Zero.copy(executionState = ExecutionState.Completed).asRight[OdbError]) { case (eDigest, atom) =>
               eDigest.flatMap: digest =>
-                digest
-                  .incrementAtomCount
-                  .filter(_.atomCount.value <= SequenceAtomLimit)
-                  .toRight(Error.sequenceTooLong(oid))
-                  .map: incDigest =>
-                    atom.steps.foldLeft(incDigest) { case (d, s) =>
-                      d.add(s.observeClass)
-                       .add(CategorizedTime.fromStep(s.observeClass, s.estimate))
-                       .add(s.telescopeConfig.offset)
-                       .copy(executionState = execState)
-                    }
+                Either.cond(digest.atomCount.value < SequenceAtomLimit, digest.add(atom), Error.sequenceTooLong(oid))
```

Comment:
Takes advantage of a change to core but is essentially unrelated to the `AtomDigest` addition.  Sorry.